File: APILunatracking/src/main/java/com/lunatrack/apilunatracking/model/CycleEntry.java
package com.lunatrack.apilunatracking.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;
import java.util.List;

@Entity
@Table(name = "cycle_entries")
@Data
public class CycleEntry {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private LocalDate date;

    private String flow;      // none, light, medium, heavy

    @ElementCollection
    @CollectionTable(name = "entry_moods")
    private List<String> moods;

    @ElementCollection
    @CollectionTable(name = "entry_symptoms")
    private List<String> symptoms;

    private String energy;    // low, medium, high
    private String sleep;     // poor, ok, good
    private String notes;
}

File: luna_track/lib/models/cycle_entry.dart
import 'package:hive/hive.dart';

part 'cycle_entry.g.dart';

@HiveType(typeId: 0)
class CycleEntry extends HiveObject {
  @HiveField(0)
  late DateTime date;

  @HiveField(1)
  late String flow; // none, light, medium, heavy

  @HiveField(2)
  late List<String> moods;

  @HiveField(3)
  late List<String> symptoms;

  @HiveField(4)
  late String energy; // low, medium, high

  @HiveField(5)
  late String sleep; // poor, ok, good

  @HiveField(6)
  late String notes;

  CycleEntry({
    required this.date,
    required this.flow,
    required this.moods,
    required this.symptoms,
    required this.energy,
    required this.sleep,
    required this.notes,
  });
}